pkg/proxy/apis/config/types.go

```diff
@@ -98,6 +98,13 @@ type KubeProxyNFTablesConfiguration struct {
 	// '1m', '2h22m'). A value of 0 means every Service or EndpointSlice change will
 	// result in an immediate iptables resync.
 	MinSyncPeriod metav1.Duration
+	// skipKernelVersionCheck causes the minimum-kernel-version check to be skipped.
+	// By default, nftables kube-proxy requires a newer kernel than is technically
+	// necessary, to prevent people from running it on older distros with older
+	// versions of the nft CLI in the host filesystem. Skipping the kernel version
+	// check will allow running nftables kube-proxy on older distros, but beware that
+	// it may interfere with the host OS's use of nftables.
+	SkipKernelVersionCheck bool
```

Comment:
Thought: @thockin @danwinship these options are kind of developer options, adding it to the config sounds a bit excessive to me, we have some places that use environment variables for similar reasons

```
staging/src/k8s.io/apimachinery/pkg/util/net/http.go:   if s := os.Getenv("DISABLE_HTTP2"); len(s) > 0 {
staging/src/k8s.io/apimachinery/pkg/util/net/http.go:   if s := os.Getenv("HTTP2_READ_IDLE_TIMEOUT_SECONDS"); len(s) > 0 {
staging/src/k8s.io/apimachinery/pkg/util/net/http.go:   if s := os.Getenv("HTTP2_PING_TIMEOUT_SECONDS"); len(s) > 0 {
```

WDYT if we use an environment variables for this instead of leaking this into the config?  
`KUBE_PROXY_NFTABLES_SKIP_VERSION_CHECK`

Reply:
I don't love it, but I see the argument.  Config feels "more serious".

Another option would be a `devMode` sub-stanza of the config?

Reply:
ping to Dan - what do you want to do with this?

Reply:
oh, I missed your earlier comment.

Having a "devMode" or "developer" subsection of the config might work, though I'm not convinced this is _only_ for developers; it's a "yes, I'm aware that I'm pointing the gun at my foot" option. It maybe could be used in production (eg, especially if the admin replaced the host nftables binary with a newer one).

Reply:
maybe env var is the best option. If we decide there are real long-term use cases for it later, we can do something different then